x-pack/plugins/security_solution/common/endpoint/data_loaders/index_fleet_agent.ts

```diff
@@ -23,7 +23,7 @@ import { wrapErrorAndRejectPromise } from './utils';
 const defaultFleetAgentGenerator = new FleetAgentGenerator();
 
 export interface IndexedFleetAgentResponse {
-  agents: Agent[];
+  agents: Array<Agent & FleetServerAgent>;
```

Comment:
Why did you have to add `FleetServerAgent` here?

Reply:
`Agent` type doesn't contain the `agent` field which `FleetServerAgent` does. United transform uses the `agent.id` field to join on so it's required. `Agent` type is actually what the fleet agents endpoint is currently typed to return but it actually does also contain the `agent` field. My guess is that when we asked fleet team to add `agent.id` to the index for the united transform recently, this api's return type wasn't updated since we were the only ones using the new field and it was via transform.

Ideally, I think we also update the api's return type but that is fleet code so I didn't want to scope creep this PR. I think it'd be best for that to be its own PR.

Reply:
Ahh. Ok. thanks for that explanation. I would suggest asking the Fleet team about this and then creating an issue to track the update if they are good with it.